src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -807,15 +818,33 @@ function discoverController(
       });
   };
 
-  $scope.updateQuery = function({ query }) {
-    $state.query = query;
+  $scope.updateQuery = function({ query, dateRange }) {
+    if (dateRange) {
+      timefilter.setTime(dateRange);
+    }
+    setAppState({ query });
     $fetchObservable.next();
   };
 
+  $scope.updateSavedQueryId = newSavedQueryId => {
+    if (newSavedQueryId) {
+      setAppState({ savedQuery: newSavedQueryId });
+    } else {
+      //reset filters and query string, remove savedQuery from state
+      const state = {
```

Comment:
The stateful search bar already handles clearing the state, so if you're subscribed to all the events, you shouldn't have to do this.

Reply:
but I've to remove the savedQuery from state/url
```TypeScript
      const state = {
        ...appStateContainer.getState(),
      };
      delete state.savedQuery;
      appStateContainer.set(state);
```
can I be sure that the state here contains no filters? (so they have been reset and state was updated before this function was executed?) just want to rule out any kind of race condition

Reply:
@lizozom should I apply this change?

Reply:
Not sure.
We can leave it for a follow up PR.